File: JS/day_11/part1.js
import { readFile } from "node:fs/promises";

async function part1(path) {
  const file = await readFile(path, "utf8");
  const data = file.trim().split("\n");

  const adj = createAdj(data);

  const dfs = (s) => {
    if (s === "out") {
      return 1;
    }

    let total = 0;

    adj[s].forEach((el) => {
      total += dfs(el);
    });

    return total;
  };

  console.log(dfs("you"));
}

function createAdj(data) {
  const adj = {};

  data.forEach((el) => {
    const parts = el.split(":");
    adj[parts[0]] = parts[1].trim().split(" ");
  });

  return adj;
}

part1("input.txt");

File: GO/day_11/part1.go
package day11

import (
	"fmt"

	"aoc2025/utils"
)

func Part1(f string) {
	data := utils.ReadStringLines(path[f])
	adj := createAdjMap(data)

	memo := make(map[string]int)
	var dfs func(s string) int
	dfs = func(s string) int {
		if val, ok := memo[s]; ok {
			return val
		}
		if s == "out" {
			return 1
		}

		total := 0

		for _, v := range adj[s] {
			total += dfs(v)
		}

		memo[s] = total
		return total
	}

	sum := dfs("you")

	fmt.Printf("Day 11 | Part 1: %d\n", sum)
}
